src/http/conn.rs

```diff
@@ -155,7 +155,8 @@ where I: Io,
     }
 
     fn maybe_park_read(&mut self) {
-        if self.io.poll_read().is_ready() {
+        // TODO: this needs fixing
```

Comment:
This was part of trying to keep from busy looping, how broken is the "this needs fixing" part :D